### PR TITLE
liquidctlfan wrapper script for your liquidctl

### PR DESCRIPTION
**Pull request by mburgholte**
As dicussed in #118 
I tried it with yoda, but it seems to be a little bit slower than sensors.

### CODE REVIEW

**Comment by jonasmalacofilho** [comment]
> I tried it with yoda, but it seems to be a little bit slower than sensors.

What you mean by slower?  Slower to respond?  If that's the case, that's intentional, as it uses an exponential moving average as a low-pass filter to avoid random spikes in fan speed and noise.

_(note: the configuration of the filter is incorrectly implemented in yoda, but it shouldn't make a huge practically difference; still, I can fix it if that's the only problem)_

---

Can I also ask you to add a (simple) README to the folder?  Your script is a very nice clean example and I would like more people to immediately get that.

**Comment by mburgholte** [comment]
You're right. Thanks. I fixed it. I'm not sure, where to put a copyright notice..

File: extra/contrib/liquidctlfan/liquidctlfan [hunk 1]
```diff
@@ -0,0 +1,94 @@
+#!/bin/bash
+# Place this script in /usr/bin or you've to adjust systemd unit file. 
+# Please have a look at cputemp. You may have to adjust something for your system.
```
Comment (jonasmalacofilho) [inline extra/contrib/liquidctlfan/liquidctlfan:3]: My suggestion would be something simple like:

```bash
#!/bin/bash
#
# liquidctlfan - wrapper script for liquidctl to control your fans
#
# Place this script in /usr/bin or you've to adjust systemd unit file. 
# Please have a look at cputemp. You may have to adjust something for your system.
#
# Copyright (C) 2020  Martin Burgholte
# SPDX-License-Identifier: GPL-3.0-or-later

Name="Liquidctl Fan Control"
...
```

**Comment by jonasmalacofilho** [comment]
> I am thinking about a separate config file?! What do you think?

If you can find a nice balance between generality and simplicity I will be very interested in seeing how you achieved that.

But I don't think it's a must here...

---

By the way, please don't forget to let me know what exactly was slow with yoda, as it's something I'll eventually get back to (more specifically, the liquidctl-integrated version of it for #72).

**Comment by mburgholte** [comment]
> > I am thinking about a separate config file?! What do you think?
> 
> If you can find a nice balance between generality and simplicity I will be very interested in seeing how you achieved that.
> 
> But I don't think it's a must here...
Sometimes it is a nice idea to have a sleep. ;o) 
Now we have all possibilities. And I think it's so easy to use.

> By the way, please don't forget to let me know what exactly was slow with yoda, as it's something I'll eventually get back to (more specifically, the liquidctl-integrated version of it for #72).

I thought I already reported the results. 
I did the following:

- time sensors
sensors  0,00s user 0,00s system 96% cpu 0,005 total

- time yoda
yoda --product 0x1711 show-sensors  0,08s user 0,02s system 2% cpu 3,841 total

**Comment by jonasmalacofilho** [comment]
Oh, I see, and that's to be expected, because yoda (as liquidctl), has to load the Python VM...

But the idea isn't to use yoda as a replacement to sensors, but rather as a replacement to a loop and successful invocations to liquidctl.

Both should be (mostly) equivalent:

```
$ liquidctlfan -p 0x1711 -u c -ct1 50.0 -ct2 60.0 -ct3 70.0 -ct4 80.0 -f0 30 -f1 40 -f2 50 -f3 80 -f4 100 -i 10
$ yoda -m grid control sync with "(49,30),(50,40),(60,50),(70,80),(100,100)" on <name from --show-sensors>
```

Yoda _should_ be lighter on the CPU because it doesn't have to initialize the Python VM, initialize the liquidctl CLI, find the correct device and connect to/disconnect from it... every iteration.  

_(That is, unless psutils, which is what yoda is using to get the hwmon data in Python, is surprising inefficient.)_

But the two commands above are not truly equivalent, because I've not passed `--interval 10`, and will default to 2 s.  In yoda I've started to implement a fancy smoothing algorithm, but it's still not smart enough to deal with very high intervals and will likely introduce too much delay to the speed adjustments.  What I mean by this delay is the time it takes from the CPU temperature to _really_ increase (that is, not considering instantaneous spikes and noise) to the the fans being properly sped up.

---

That said, your script is a work a scripting art!

**Pull request merged**